stake-pool/single-pool/tests/deposit.rs

```diff
@@ -0,0 +1,453 @@
+    solana_program_test::*,
+    solana_sdk::{
+        instruction::InstructionError,
+        message::Message,
+        signature::Signer,
+        signer::keypair::Keypair,
+        stake::{
+            instruction::StakeError,
+            state::{Authorized, Lockup},
+        },
+        transaction::Transaction,
+    },
+    spl_associated_token_account as atoken,
+    spl_single_validator_pool::{
+        error::SinglePoolError, find_default_deposit_account_address, id, instruction,
+    },
+    test_case::test_case,
+};
+
+#[test_case(true, 0, false; "activated")]
+#[test_case(false, 0, false; "activating")]
+#[test_case(true, 100_000, false; "activated_extra")]
+#[test_case(false, 100_000, false; "activating_extra")]
+#[test_case(true, 0, true; "activated_second")]
+#[test_case(false, 0, true; "activating_second")]
+#[tokio::test]
+async fn success(activate: bool, extra_lamports: u64, prior_deposit: bool) {
+    let mut context = program_test().start_with_context().await;
+    let accounts = SinglePoolAccounts::default();
+    accounts
+        .initialize_for_deposit(
+            &mut context,
+            TEST_STAKE_AMOUNT,
+            if prior_deposit {
+                Some(TEST_STAKE_AMOUNT * 10)
+            } else {
+                None
+            },
+        )
+        .await;
+
+    if activate {
+        advance_epoch(&mut context).await;
+    }
+
+    if prior_deposit {
+        let instructions = instruction::deposit(
+            &id(),
+            &accounts.vote_account.pubkey(),
+            &accounts.bob_stake.pubkey(),
+            &accounts.bob_token,
+            &accounts.bob.pubkey(),
+            &accounts.bob.pubkey(),
+        );
+        let message = Message::new(&instructions, Some(&accounts.bob.pubkey()));
+        let transaction = Transaction::new(&[&accounts.bob], message, context.last_blockhash);
+
+        context
+            .banks_client
+            .process_transaction(transaction)
+            .await
+            .unwrap();
+    }
+
+    let wallet_lamports_after_stake =
+        get_account(&mut context.banks_client, &accounts.alice.pubkey())
+            .await
+            .lamports;
+
+    let (_, alice_stake_before_deposit, stake_lamports) =
+        get_stake_account(&mut context.banks_client, &accounts.alice_stake.pubkey()).await;
+    let alice_stake_before_deposit = alice_stake_before_deposit.unwrap().delegation.stake;
+
+    let (_, pool_stake_before, pool_lamports_before) =
+        get_stake_account(&mut context.banks_client, &accounts.stake_account).await;
+    let pool_stake_before = pool_stake_before.unwrap().delegation.stake;
+
+    if extra_lamports > 0 {
+        transfer(
+            &mut context.banks_client,
+            &context.payer,
+            &context.last_blockhash,
+            &accounts.stake_account,
+            extra_lamports,
+        )
+        .await;
+    }
+
+    let mut fees = USER_STARTING_LAMPORTS - wallet_lamports_after_stake - stake_lamports;
```

Comment:
If you don't want to worry about fees, how about setting `context.payer` as the fee payer?